examples/mysql-atomic-transaction/mysql_atomic_transaction.bal

```diff
@@ -13,23 +13,14 @@ public function main() returns error? {
     transaction {
         _ = check mysqlClient->execute(`INSERT INTO Customers (firstName, lastName, registrationID, creditLimit,
                                         country) VALUES ('Linda', 'Jones', 4, 10000.75, 'USA')`);
-        log:printInfo("First query executed successfully.");
 
-        // Insert a `customer` record, which violates the unique
-        sql:ExecutionResult|sql:Error result = mysqlClient->execute(
-                `INSERT INTO Customers (firstName, lastName, registrationID, creditLimit, country)
-                                VALUES ('Peter', 'Stuart', 4, 5000.75, 'USA')`);
+        _ = check mysqlClient->execute(`INSERT INTO Customers (firstName, lastName, registrationID, creditLimit,
+                                        country) VALUES ('Peter', 'Stuart', 4, 5000.75, 'USA')`);
 
-        if result is sql:Error {
-            log:printError(result.message());
-            log:printInfo("Second query failed. Rollback transaction.");
-            rollback;
-        } else {
-            error? err = commit;
-            if err is error {
-                log:printError("Error occurred while committing", err);
-            }
-        }
+        check commit;
+    } on fail error e {
+        log:printError(e.message());
+        log:printInfo("One of the queries failed. Rollback transaction.");
```

Comment:
Don't we need to rollback here

Reply:
Rollback is called automatically if the statements return error from check statements